graphql/resolvers.go

```diff
@@ -1502,37 +1502,31 @@ func (r *queryResolver) Projects(ctx context.Context) ([]*GroupedProjects, error
 		return nil, ResourceNotFound.Send(ctx, err.Error())
 	}
 
-	groupsMap := make(map[string][]*restModel.APIProjectRef)
-
-	for _, p := range allProjs {
-		groupName := strings.Join([]string{p.Owner, p.Repo}, "/")
-		apiProjectRef := restModel.APIProjectRef{}
-		if err = apiProjectRef.BuildFromService(p); err != nil {
-			return nil, InternalServerError.Send(ctx, fmt.Sprintf("error building APIProjectRef from service: %s", err.Error()))
-		}
-
-		if projs, ok := groupsMap[groupName]; ok {
-			groupsMap[groupName] = append(projs, &apiProjectRef)
-		} else {
-			groupsMap[groupName] = []*restModel.APIProjectRef{&apiProjectRef}
-		}
+	groupedProjects, err := GroupProjects(allProjs, false)
```

Comment:
sorry if i've asked this before: what is this resolver used for?

Reply:
It's used here: 
https://github.com/evergreen-ci/spruce/blob/main/src/pages/Commits.tsx#L131

Reply:
Ah is this for the mainline commits dropdown or?

Reply:
It's part of the mainline commits ui [EVG-13855](https://jira.mongodb.org/browse/EVG-13855) @khelif96 can you maybe provide more details?

Reply:
Its used to fetch the projects for the project selector on the waterfall. And will also be added to the projects patches page in the near future.

Reply:
I do think we should incorporate the logic chaya added to determine what projects a user can view. Otherwise this just returns all the projects available in evergreen

Reply:
Chaya's logic shows what projects a user can view _settings_ for. Theoretically this should return just the projects that a user has permission to see (tasks view permission) but I imagine that should be a separate ticket.